manifests/init.pp

```diff
@@ -185,6 +182,7 @@
   $enable_pulp2_deb = $enable_deb and !($pulpcore and $proxy_pulp_deb_to_pulpcore)
 
   $foreman_url = $foreman_proxy::foreman_base_url
+  $foreman_hostname = regsubst($foreman_url, 'https://', '')
```

Comment:
The Foreman URL can really be a URL and include a path. I think the safest would be to add a function to extract the hostname from a URL. https://github.com/voxpupuli/puppet-extlib/blob/master/lib/puppet/functions/extlib/dump_args.rb is probably the best example of a minimal function. Then use Ruby's URL parsing to do so.

Reply:
Hah. I almost went that route. Then said... let me try the 'dumb' route and save myself time just in case :)

Reply:
I wonder if that's actually true for Katello deployments, or if we hardcore paths like /rhsm on clients so that you can't deploy it in a "folder".

Reply:
Still, writing your own parsing is a likely a source of problems. For example, even if we only support `/`, the current code doesn't deal with `https://foreman.example.com/` which is a valid input.